### PR TITLE
Updating GetAtt values.

### PR DESCRIPTION
**Pull request by jonjara**
*Issue #, if available:* #41 

*Description of changes:*

These are the values for `GetAtt` 

Domain
```
!GetAtt MyDomain.Name
!GetAtt MyDomain.Owner
!GetAtt MyDomain.EncryptionKey
````

Repo: 
```
!GetAtt MyRepo.Name
!GetAtt MyRepo.DomainName
!GetAtt MyRepo.DomainOwner
```

**Testing**

Gave this template a whirl 

```
Parameters:
  Name:
    Type: String
  DomainName:
    Type: String    

Resources:
  TestCodeArtifactDomain:
    Type: 'AWS::CodeArtifact::Domain'
    Properties:
      DomainName: !Ref DomainName

  TestCodeArtifacRepository:
    Type: 'AWS::CodeArtifact::Repository'
    Properties:
      RepositoryName: !Ref Name
      DomainName: !GetAtt TestCodeArtifactDomain.Name  

Outputs:
  RepoArn:
    Value: !Ref TestCodeArtifacRepository

  DomainArn:
    Value: !Ref TestCodeArtifactDomain    

  RepoArnWithGetAtt:
    Value: !GetAtt TestCodeArtifacRepository.Arn   

  RepoDomainWithGetAtt:
    Value: !GetAtt TestCodeArtifactDomain.Arn        

  RepoNameFromRepo:
    Value: !GetAtt TestCodeArtifacRepository.Name    

  DomainNameFromDomain:
    Value: !GetAtt TestCodeArtifactDomain.Name   

  DomainOwnerFromDomain:
    Value: !GetAtt TestCodeArtifactDomain.Owner            

  DomainNameFromRepo:
    Value: !GetAtt TestCodeArtifacRepository.DomainName  

  DomainOwnerFromRepo:
    Value: !GetAtt TestCodeArtifacRepository.DomainOwner          
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/UpdateHandler.java [hunk 1]
```diff
@@ -24,11 +23,14 @@
...
-        String previousDomainName = request.getPreviousResourceState().getDomainName();
-        if (!Objects.equals(previousDomainName, desiredDomainName)) {
-            // cannot update domainName because it's CreateOnly
-            throw new CfnNotUpdatableException(ResourceModel.TYPE_NAME, desiredDomainName);
+        ResourceModel desiredResourceState = request.getDesiredResourceState();
+        ResourceModel previousResourceState = request.getPreviousResourceState();
+        if (!Objects.equals(previousResourceState.getDomainName(), desiredResourceState.getDomainName()) ||
+            !Objects.equals(previousResourceState.getEncryptionKey(), desiredResourceState.getEncryptionKey())
+        ) {
+            logger.log("Unupdatable exception here jonjar");
```
Comment (sankalpbhandari) [inline aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/UpdateHandler.java:31]: Nit: Can you update the log

**Review by sankalpbhandari — CHANGES_REQUESTED** [review]

File: aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/UpdateHandler.java (hunk 1 above)
Comment (jonjara) [reply aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/UpdateHandler.java:31]: lol

**Review by sankalpbhandari — APPROVED** [review]

File: aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/Translator.java [hunk 2]
```diff
@@ -58,12 +58,9 @@ static CreateDomainRequest translateToCreateRequest(final ResourceModel model) {
   /**
    * Request to read a resource
    * @param model resource model
-   * @param request resource handler request
    * @return awsRequest the aws service request to describe a resource
    */
-  static DescribeDomainRequest translateToReadRequest(
-      final ResourceModel model, final ResourceHandlerRequest<ResourceModel> request
```
Comment (jonjara) [inline aws-codeartifact-domain/src/main/java/software/amazon/codeartifact/domain/Translator.java:65]: `request` was not being used, doesn't really have to do with the description of the PR

File: aws-codeartifact-domain/src/test/java/software/amazon/codeartifact/domain/ReadHandlerTest.java [hunk 3]
```diff
@@ -97,6 +97,8 @@ public void handleRequest_simpleSuccess() {
         final ResourceModel desiredOutputModel = ResourceModel.builder()
             .domainName(DOMAIN_NAME)
             .domainOwner(DOMAIN_OWNER)
+            .name(DOMAIN_NAME)
+            .owner(DOMAIN_OWNER)
```
Comment (detam-amazon) [inline aws-codeartifact-domain/src/test/java/software/amazon/codeartifact/domain/ReadHandlerTest.java:101]: Do we need an explanation here of why we're storing the same information twice? (domainOwner + owner, and domainName + name)

Comment (joeymarylander) [reply aws-codeartifact-domain/src/test/java/software/amazon/codeartifact/domain/ReadHandlerTest.java:101]: There is an explanation in the model json itself. It says that those new fields are used in GetAtt. Not sure if we need to explain more than that

**Review by joeymarylander — APPROVED** [review]
lgtm

File: aws-codeartifact-domain/src/test/java/software/amazon/codeartifact/domain/ReadHandlerTest.java (hunk 3 above)
Comment (detam-amazon) [reply aws-codeartifact-domain/src/test/java/software/amazon/codeartifact/domain/ReadHandlerTest.java:101]: Thanks

Comment (jonjara) [reply aws-codeartifact-domain/src/test/java/software/amazon/codeartifact/domain/ReadHandlerTest.java:101]: I did some testing and we may not need to have both for reading. We need them both to be present in the model but we don't need both unless we want both `DomainName` and `Name` to be available to `GetAtt`

**Pull request merged**